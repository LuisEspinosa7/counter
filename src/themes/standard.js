export const theme = {
    colors: {
        primary: {
            normal: '#4764e6',
            light: '#398AB9',
            dark: '#145071'
        },
        secondary: {
            normal: '#f2f4ff'
        },
        accent: {
            normal: '#4bd6f2'
        },
        gray: {
            normal: '#EEEEEE',
            dark: '#e6e4e4'
        },
        white: {
            normal: '#fff'
        },
        black: {
            normal: '#000'
        }
    },
    mainContainer: {
        maxWidth: '1600px'
    }
}