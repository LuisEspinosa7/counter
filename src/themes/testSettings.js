import { ThemeConsumer } from 'styled-components';
import { shallow } from 'enzyme/build';
import { theme } from './standard';


/* This is an approach to solve the enzyme tests when using a
   Styled components Theme
 */
export const shallowWithTheme = (children, them = theme) => {
    ThemeConsumer._currentValue = them;
    return shallow(children);
}