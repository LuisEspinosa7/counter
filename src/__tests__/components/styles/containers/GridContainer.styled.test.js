import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../../themes/standard';
import { render } from 'enzyme/build';
import { StyledGridContainer } from '../../../../components/styles/containers/GridContainer.styled';


describe("Component: StyledGridContainer", () => {

    it('should render', () => {
        const { fragment } = render(
            <ThemeProvider theme={theme}>
                <StyledGridContainer />
            </ThemeProvider>
        )
        const tree = renderer.create(fragment).toJSON();
        expect(tree).toMatchSnapshot();
    })

})