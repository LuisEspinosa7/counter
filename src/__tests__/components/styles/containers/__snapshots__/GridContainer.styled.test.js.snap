// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: StyledGridContainer should render 1`] = `null`;
